Source/ACE/Network/NetworkSession.cs

```diff
@@ -164,59 +191,149 @@ private void ProcessPacket(ClientPacket packet)
                 // We will send this at a 20 second time interval.  I don't know what to do with these when we receive them at this point.
             }
 
+            // If the client is requesting a retransmission, pull those packets from the queue and resend them.
             if (packet.Header.HasFlag(PacketHeaderFlags.RequestRetransmit))
             {
                 foreach (uint sequence in packet.HeaderOptional.RetransmitData)
                     Retransmit(sequence);
             }
 
+            // This should be set on the first packet to the server indicating the client is logging in.
+            // This is the start of a three-way handshake between the client and server (LoginRequest, ConnectRequest, ConnectResponse)
+            // Note this would be sent to each server a client would connect too (Login and each world).
+            // In our current implimenation we handle all roles in this one server.
             if (packet.Header.HasFlag(PacketHeaderFlags.LoginRequest))
             {
                 AuthenticationHandler.HandleLoginRequest(packet, session);
                 return;
             }
 
+            // This should be set on the second packet to the server from the client.
+            // This comletes the three-way handshake.
             if (packet.Header.HasFlag(PacketHeaderFlags.ConnectResponse))
             {
                 sendResync = true;
                 AuthenticationHandler.HandleConnectResponse(packet, session);
                 return;
             }
 
+            // Process all fragments out of the packet
             foreach (ClientPacketFragment fragment in packet.Fragments)
                 ProcessFragment(fragment);
 
+            // Update the last received sequence.
             if (packet.Header.Sequence != 0)
-                lastReceivedSequence = packet.Header.Sequence;
+                lastReceivedPacketSequence = packet.Header.Sequence;
         }
 
+        /// <summary>
+        /// Processes a fragment, combining split fragments as needed, then handling them
+        /// </summary>
+        /// <param name="fragment">ClientPacketFragment to process</param>
         private void ProcessFragment(ClientPacketFragment fragment)
         {
+            log.DebugFormat("[{0}] Processing fragment {1}", session.Account, fragment.Header.Sequence);
+            ClientMessage message = null;
+            // Check if this fragment is split
             if (fragment.Header.Count != 1)
             {
-                lock (partialFragments)
+                // Packet is split
+                log.DebugFormat("[{0}] Fragment {1} is split, this index {2} of {3} fragments", session.Account, fragment.Header.Sequence, fragment.Header.Index, fragment.Header.Count);
+                MessageBuffer buffer = null;
+
+                // partialFragments is only ever used here in the main update loop, so no need to lock.

```

Comment:
This is not true, need to lock this actually, as this is coming in on ProcessPacket